src/app/page.tsx

```diff
@@ -1,13 +1,100 @@
+"use client";
 import Image from "next/image";
+import { useEffect, useState } from "react";
+import { abi } from "./utils/abi";
+import { useSearchParams } from "next/navigation";
+import { ConnectButton } from "@rainbow-me/rainbowkit";
+import {
+  useAccount,
+  useWriteContract,
+  useWaitForTransactionReceipt,
+} from "wagmi";
+import { checkAndRequestSFuel } from "./utils/sfuelClaim";
+import { SyncLoader } from "react-spinners";
+import toast, { Toaster } from "react-hot-toast";
+import { performMulticall } from "./multicall";
 
 export default function Home() {
+  const [isClaiming, setIsClaiming] = useState(false);
+  const MIN_SFUEL_BALANCE = BigInt(10 ** 16);
+  const { address, isConnected } = useAccount();
+  const searchParams = useSearchParams();
+  const searchHash =
+    searchParams.get("hash") ||
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
You could use the zeroAddress constant by viem rather than hardcoding the value. Also if there is no valid hash in the params, then I should ideally not be able to click the claim button